pyqtgraph/tests/image_testing.py

```diff
@@ -306,15 +309,25 @@ def saveFailedTest(data, expect, filename):
     img[2:2+diff.shape[0], -diff.shape[1]-2:-2] = diff
 
     png = makePng(img)
+    directory, _, _ = filename.rpartition("/")
+    if not os.path.isdir(directory):
+        os.makedirs(directory)
+    with open(filename + ".png", "wb") as png_file:
+        png_file.write(png)
+    print("\nImage comparison failed. Test result: %s %s   Expected result: "
+        "%s %s" % (data.shape, data.dtype, expect.shape, expect.dtype))
+    if upload:
+        uploadFailedTest(filename)
     
+def uploadFailedTest(filename):
```

Comment:
I guess this is soon-to-be dead code anyway, but `png` is not defined in this function.

Reply:
you're right; I need to pass the png in there; also regarding the dead code, I'll leave that up to you, as you are the one maintaining the server.

Reply:
fixed with c4e295c